src/05-derivatives/02-common-data-types.md

```diff
@@ -62,9 +71,50 @@ makes them invalid (e.g., if a source 4D image is averaged to create a single
 static volume, a SamplingFrequency property would no longer be relevant). In
 addition, all processed files include the following metadata JSON fields:
 
-| **Key name**  | **Description**                                                                                 |
-| ------------- | ----------------------------------------------------------------------------------------------- |
-| SkullStripped | REQUIRED. Boolean. Whether the volume was skull stripped (non-brain voxels set to zero) or not. |
+| **Key name**  | **Description**                                                                                                              |
+| ------------- | ---------------------------------------------------------------------------------------------------------------------------- |
+| SkullStripped | REQUIRED. Boolean. Whether the volume was skull stripped (non-brain voxels set to zero) or not.                              |
+| Resolution    | REQUIRED if `res` keyword present. Dictionary of Indexes to Strings. Specifies the interpretation of the resolution keyword. |
+| Density       | REQUIRED if `den` keyword present. Dictionary of Strings to Strings. Specifies the interpretation of the resolution keyword. |
```

Comment:
Our usual way to disambiguate entities is something like:

`res-1_bold.json`:
```
{
  "Resolution": "abc"
}
```

`res-2_bold.json`:
```
{
  "Resolution": "xyz"
}
```

The switch to a dictionary indexed by strings doesn't appear to be justified anywhere in the proposed text. And for resolution, if it's an index, should it not be an integer, rather than a string of digits?

Reply:
I'm very happy to elaborate more on this:

- Why an index: to allow developers to impose an implicit hierarchy. However, I would agree with alternatives such as `res-high`, `res-low`, `res-mid`... but that vocabulary would be very limited. I wouldn't allow `res-2x2x2` for various reasons. First is that very often we will need decimals and `res-07x07x07` seems like a recipe for disaster. Second, file formats generally address this issue, so it gives plenty of room for the discrepancy between this value and what you find in the file's header. To me, the index solution seemed easy.

However, to allow human-readable descriptions in the JSON file, they need to be converted to string indexes for the dictionary (JSON does not allow integer keys, does it?).

I can add your metadata proposal, but I would not replace the dictionary because that allows you to place the metadata at a higher level of the hierarchy (ie., describe res-1 and res-2 in the same, unique file).

- I like the idea of free-form descriptions of each index. The rationale is that - software should read the actual resolution (and units that seem to be of concern) from the file format. Humans, however, might check quickly this description field without the need of opening the file (e.g., `nib-ls` for niftis with nibabel). I would be very happy to change my mind if a better idea is proposed, or improving this PR with a better description of the current proposal - which seems necessary at this point.